habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -0,0 +1,776 @@
+        if state["robot_T"] is not None:
+            self.robot.sim_obj.transformation = state["robot_T"]
+            self.robot.sim_obj.clear_joint_states()
+
+        if "robo_js" in state:
+            self.robot.arm_joint_pos = state["robo_js"]
+
+        for T, ao in zip(state["art_T"], self.art_objs):
+            ao.transformation = T
+
+        for T, i in zip(state["static_T"], self.scene_obj_ids):
+            self.reset_obj_T(i, T)
+
+        for p, ao in zip(state["art_pos"], self.art_objs):
+            ao.joint_positions = p
+
+        if set_hold:
+            if state["obj_hold"] is not None:
+                self.internal_step(-1)
+                self.full_snap(self.scene_obj_ids.index(state["obj_hold"]))
+            elif state["marker_hold"] is not None:
+                self.set_snapped_marker(state["marker_hold"])
+            else:
+                self.desnap_object(True)
+
+    def reset_obj_T(self, i, T):
+        self.set_transformation(T, i)
+        self.set_linear_velocity(mn.Vector3(0, 0, 0), i)
+        self.set_angular_velocity(mn.Vector3(0, 0, 0), i)
+
+    def reset_art_obj_pos(self, i, p):
+        self.set_articulated_object_positions(i, p)
+        vel = self.get_articulated_object_velocities(i)
+        forces = self.get_articulated_object_forces(i)
+        self.set_articulated_object_velocities(i, np.zeros((len(vel),)))
+        self.set_articulated_object_forces(i, np.zeros((len(forces),)))
+
+    def settle_sim(self, seconds):
+        steps = int(seconds * self.ctrl_freq)
+        for _ in range(steps):
+            self.step_world(-1)
+
+    def full_snap(self, obj_id):
+        """
+        No matter where the object is in the scene, it will be moved to the
+        robot's gripper and snapped. This should be used for "teleporting" the
+        object to the robot's hand, it is not physically plausible.
+        - obj_id: The index in scene_obj_ids, not absolute simulator index.
+        """
+        abs_idx = self.scene_obj_ids[obj_id]
+        ee_T = self.get_end_effector_trans()
+        obj_local_T = mn.Matrix4.translation(EE_GRIPPER_OFFSET)
+        global_T = ee_T @ obj_local_T
+        self.set_transformation(global_T, abs_idx)
+        self.set_snapped_obj(obj_id)
+
+    def set_snapped_obj(self, snapped_obj_id):
+        """
+        - snapped_obj_id: the index of the object in scene_obj_ids. Not
+          absolute simulator index.
+        """
+        use_snap_obj_id = self.scene_obj_ids[snapped_obj_id]
+        if use_snap_obj_id == self.snapped_obj_id:
+            return
+        if len(self.snapped_obj_constraint_id) != 0:
+            self.desnap_object()
+
+        self.snapped_obj_id = use_snap_obj_id
+
+        max_impulse = 1000.0
+        if self.snap_to_link_on_grab:
+            # Set collision group to GraspedObject so that it doesn't collide
+            # with the links of the robot.
+            grasped_object_group = 16  # see Habitat-sim CollisionGroupHelper.h
```

Comment:
The default groups have changed somewhat, be sure you have these configured correctly.